packages/react-dogfood/pages/guest/join/[guestCallId].tsx

```diff
@@ -37,21 +37,11 @@ export default function GuestCallRoom(props: GuestCallRoomProps) {
   );
   const [tokenToUse] = useState(mode === 'anon' ? token : undefined);
   const [client] = useState<StreamVideoClient>(
-    () => new StreamVideoClient(apiKey),
+    () =>
+      new StreamVideoClient({ apiKey, user: userToConnect, token: tokenToUse }),
   );
```

Comment:
I've seen this pattern here: https://codesandbox.io/s/stream-video-quickstart-forked-2sui2h?file=/src/App.tsx:1048-1096, do you usually use the `useMemo` in these cases?

Reply:
Depends on the use case, I'm not sure if this pattern was used to keep the references to either call or client stable. In that case I'd probably reach for `useRef`, `useState` is usually used to imperatively update the value with `setState` (to trigger re-render and effects). But I'm guessing that `userToConnect`, `tokenToUse` (and other) can change over time making these stateful values stale.

Reply:
`useMemo` then would keep the reference stable as long as none of the dependencies change.

Reply:
I tested and `userToConnect` and `tokenToUse ` can't change (but I might be missing something), so I think the code should be ok, but it might be that it's not best practice to use `useState`. I can rewrite it using `useMemo`, I guess that's kind of the safest option as you can list the dependencies there.

Reply:
I don't see `setState` being utilised anywhere so esentially it is the same as:

```ts
const client = useMemo(() => new StreamVideoClient({ apiKey, user: userToConnect, token: tokenToUse }), []);

useEffect(() => {
    // do whatever
    return () => {
        // do cleanup
    }
}, [client])
```

Reply:
@arnautov-anton by stateful you mean, that this (client instance) could be a constant outside the component?